### PR TITLE
persist wx_armor & wait for arm to come online instead of crashing

### PR DESCRIPTION
**Pull request by le-horizon**
as title.  Tested on wx deployment app by unplugging and plugging power.

### CODE REVIEW

File: wx_armor/wx_armor/wx_armor_ws.h [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 namespace horizon::wx_armor {
 
-WxArmorDriver *Driver();
+WxArmorDriver *Driver(bool force_init = false);
```
Comment (QuantuMope) [inline wx_armor/wx_armor/wx_armor_ws.h:19]: When is `force_init` ever used?

**Review by QuantuMope — COMMENTED** [review]
Hey Le had one comment.

Also, will this mean wxarmor will continuously be checking connection throughout deployment? 

Wondering if we can have some kind of timeout. Or maybe since it's a single ping call every 3 seconds it's okay. Not sure.

File: wx_armor/wx_armor/wx_armor_ws.h (hunk 1 above)
Comment (le-horizon) [reply wx_armor/wx_armor/wx_armor_ws.h:19]: it's used in the cc file to force reinitialize driver.  This is needed when wx powers off and then on.

**Comment by le-horizon** [comment]
> Hey Le had one comment.
> 
> Also, will this mean wxarmor will continuously be checking connection throughout deployment?

It will only look for the wx device when it cannot find it.  Once it's connected, it will not look for wx.

> 
> Wondering if we can have some kind of timeout. Or maybe since it's a single ping call every 3 seconds it's okay. Not sure.

**Comment by QuantuMope** [comment]
> It will only look for the wx device when it cannot find it. Once it's connected, it will not look for wx.

Right, I meant more in the scenario where we never use wx. For example, operating the z1.

File: wx_armor/wx_armor/wx_armor_ws.h (hunk 1 above)
Comment (QuantuMope) [reply wx_armor/wx_armor/wx_armor_ws.h:19]: When I do a quick search, I never see any `Driver(true)` calls.

Comment (le-horizon) [reply wx_armor/wx_armor/wx_armor_ws.h:19]: ha, you are right.  Looks like the driver can handle disconnect and then connect of the wx arm without re-instantiation.

Comment (le-horizon) [reply wx_armor/wx_armor/wx_armor_ws.h]: Reverted.

**Comment by le-horizon** [comment]
> > It will only look for the wx device when it cannot find it. Once it's connected, it will not look for wx.
> 
> Right, I meant more in the scenario where we never use wx. For example, operating the z1.

For Z1, we just do not start wx_armor &, and no resources will be wasted.

**Pull request merged**